sky/packages/sky/lib/src/rendering/paragraph.dart

```diff
@@ -59,22 +59,27 @@ class RenderParagraph extends RenderBox {
     textPainter.minHeight = constraints.minHeight;
     textPainter.maxHeight = constraints.maxHeight;
     textPainter.layout();
+    // By default, we shrinkwrap to the intrinsic width.
+    double width = constraints.constrainWidth(textPainter.maxIntrinsicWidth);
```

Comment:
> What width were we using before that was causing the text to be off-screen when right-aligned?

We'd layout the text with the max width we got from our parents and then we'd use our intrinsic width as our width.  When right-aligned, we paint the first letter at the max width to the right of our left edge.